bundles/org.openhab.binding.novafinedust/src/main/java/org/openhab/binding/novafinedust/internal/SDS011Handler.java

```diff
@@ -0,0 +1,266 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "Port is in use!");
+            return;
+        } catch (TooManyListenersException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Cannot attach listener to port!");
+            return;
+        } catch (UnsupportedCommOperationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Cannot set serial port parameters");
+            return;
+        }
+
+        if (initSuccessful) {
+            lastCommunication = ZonedDateTime.now();
+            updateStatus(ThingStatus.ONLINE);
+
+            if (mode == WorkMode.POLLING) {
+                pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                    try {
+                        communicator.requestSensorData();
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                "Cannot query data from device");
+                    }
+                }, 2, config.pollingInterval, TimeUnit.SECONDS);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Commands and replies from the device don't seem to match");
+            logger.debug("Could not configure sensor -> setting Thing to OFFLINE and disposing the handler");
+            dispose();
+        }
+    }
+
+    private boolean validateConfiguration() {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Configuration could not be parsed");
+            return false;
+        }
+
+        if (config.port == null || config.port.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Port must be set!");
+            return false;
+        }
+        if (config.reporting) {
+            if (config.reportingInterval < 0 || config.reportingInterval > 30) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "Reporting interval has to be between 0 and 30 minutes");
+                return false;
+            }
+        } else {
+            if (config.pollingInterval < 3 || config.pollingInterval > 3600) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "Polling interval has to be between 3 and 3600 seconds");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
```

Comment:
Another example here of how to easily deal with the null warnings.
```suggestion
        Future<?> pollingJob = this.pollingJob;
        if (pollingJob != null) {
            pollingJob.cancel(true);
            this.pollingJob = null;
        }
```

Reply:
My IDE wasn't even showing this warning.

So to me it looks like the IDE checks and the build checks are not in sync. There was even a build failure in travis (a couple of commits back) which was fine in eclipse.

Also If you insists of dealing with these warnings, I would recommend to make them errors, not warnings, so the build will fail direclty.

One side note:

If you implement is as you suggested, you will hide the class variable `pollingJob` with the local variable with the same name and this is a bad code style which can lead to errors. Even Sonoarlint marks this as something that should be changed.